### PR TITLE
UserService_EmailSender_AddLogsAndException

### PR DESCRIPTION
**Pull request by strannick-rodion**

### CODE REVIEW

File: Services/UserService/Utils/EmailSender.cs [hunk 1]
```diff
@@ -33,19 +34,25 @@ public void SendEmail(string email, ISecretTokenEngine secretTokenEngine)
...
                 }
-                catch (SmtpException)
+                catch (SmtpException e)
                 {
-                    Thread.Sleep(10000 * k);
+                    Thread.Sleep(10000 * (++i));
                     flag = false;
-                    if (k < 5)
-                        k++;
+                    logger.LogWarning(e, "SmtpException thrown while trying to Send Eamil to confirm");
```
Comment (nmamro) [inline Services/UserService/Utils/EmailSender.cs]: It's not formatted for correlation

File: Services/UserService/Utils/EmailSender.cs [hunk 2]
```diff
@@ -21,7 +22,7 @@ public void SendEmail(string email, ISecretTokenEngine secretTokenEngine)
             var from = new MailAddress("t.platform@mail.ru", "Trading Station");
             var to = new MailAddress(email);
             string secretToken = secretTokenEngine.GetToken(email).ToString();
-            string link = $"https://localhost:44335//confirm/{secretToken}";
+            string link = $"https://localhost:44335/confirm/{secretToken}";
```
Comment (denis-kriv) [inline Services/UserService/Utils/EmailSender.cs:25]: We will use http

File: Services/UserService/Utils/EmailSender.cs [hunk 3]
```diff
@@ -33,19 +34,25 @@ public void SendEmail(string email, ISecretTokenEngine secretTokenEngine)
...
                 {
                     flag = true;
                     smtp.Send(m);
+                    break;
                 }
-                catch (SmtpException)
+                catch (SmtpException e)
                 {
-                    Thread.Sleep(10000 * k);
+                    Thread.Sleep(10000 * (++i));
```
Comment (denis-kriv) [inline Services/UserService/Utils/EmailSender.cs:47]: why not to use `Thread.Sleep(10000)`

File: Services/UserService/Utils/EmailSender.cs [hunk 4]
```diff
@@ -33,19 +34,25 @@ public void SendEmail(string email, ISecretTokenEngine secretTokenEngine)
 
             var flag=true;
```
Comment (denis-kriv) [inline Services/UserService/Utils/EmailSender.cs:35]: spaces

Comment (denis-kriv) [reply Services/UserService/Utils/EmailSender.cs:35]: And do we need flag?

File: Services/UserService/Utils/EmailSender.cs (hunk 3 above)
Comment (strannick-rodion) [reply Services/UserService/Utils/EmailSender.cs:47]: Because we wont to have more time after every try

**Review by Cerberous007 — APPROVED** [review]

**Pull request merged**